### ISSUE TITLE
How to run a Propagation or Router node?

### ISSUE THREAD

**Issue by trosel** [markqvist/LXMF#6]
It's not clear who runs these nodes that are seemingly required to ensure reliable communication (in cases where destinations are offline at the time of sending).

If I run the Nomad application, does that automatically use my own device as my own router?

**Comment by markqvist** [comment]
The information on all of this is included in the guide section of the Nomad Network program, which is admittedly a bit hard to find, if you are just looking at it here from GitHub.

If you enable node hosting in the Nomad Network, it will also act as a propagation node. This is currently the only way to provide propagation nodes on a Reticulum network. 

The next release of LXMF will include a separate `lxmd` program that can run a propagation node as a daemon.

Nomad Network will automatically use the nearest *trusted* node as a propagation node. Or you can manually specify a specific node to use in the program. Sideband will use the nearestnearest available node, or the specific node configured in its settings.

**Comment by trosel** [comment]
If the app finds the nearest node, what is the privacy concern there? Are you exposing your IP to every node? It has forward secrecy so Id imagine it is quite privacy preserving.

**Comment by markqvist** [comment]
Any connections to any nodes happen over Reticulum, not IP directly, although your Reticulum link could run over one or more IP connections, that might expose the IP address you are communicating from (if using IP in the first place).

That being said, the link from the sender to the propagation node is always anonymous. The sender does not identify. And when the message is being passed on to the propagation node, it is already encrypted, which means origin address is hidden.

A malicious Propagation Node operator can only learn that *someone* has sent a message to a given address. No information about where the message originally came from is visible.

I will get the ful documentation about LXMF up soon. Hopefully this cleared it up a little in the meantime!

**Comment by trosel** [comment]
> A malicious Propagation Node operator can only learn that someone has sent a message to a given address. No information about where the message originally came from is visible.

If that is true, it makes it very difficult to ever economically incentivize the running of a node, right?

You couldn’t perform a check to see if the origin sender has paid for usage or anything

**Comment by markqvist** [comment]
That is correct, and this is decision is an integral choice of the design. There would simply be no way to provide any reasonable level of privacy or anonymity if you can match on sender identities throughout the network. Therefore they remain completely hidden, and are only revealed to who the sender wants to reveal it to.

If you want to run propagation nodes as a commercial offering, there are other models to allow access only to paying customers, that are compatible with the privacy-first design of this system.

But it don't necessarily think the best place to put your bets, in terms of financial profit, would be the running of propagation nodes. It is so ridiculously cheap to run a PN that you would most likely be "outcompeted" by people just making putting them on the network for their own purposes, and also making them available at large.

I designed the system so it would be beneficial to individuals and organisations to run their own PNs, while making it practically costless and also easier to just let them help out the rest of the network instead of trying to lock them down to specific users.

**Comment by trosel** [comment]
I *love* that the design includes forward secrecy from the very beginning. It's like putting your messages in one of these:

![image](https://user-images.githubusercontent.com/1299322/167184165-a3a25c35-3d04-49e3-8a5b-99624ff2041d.png)

Mailman doesn't care who it's from, just where to take it.

> If you want to run propagation nodes as a commercial offering, there are other models to allow access only to paying customers, that are compatible with the privacy-first design of this system.

Out of curiousity, what is that? I can't help but think of someone running a node in a co-op community kind of way. There are just too many people that would prefer to not have to think about maintaining servers.

**Comment by markqvist** [comment]
Yes, exactly! That is a very fitting comparison :)

The best incentive for people to run propagation nodes (or any kind of shared infrastructure) is that it actually *helps themselves first*, while being, in practical term, costless. If you already have some sort of system running a Reticulum transport node, you might as well run a propagation node on it, for your own sake. This will make sure you always have instant access to all the messages/information you need. This will also help out the rest of the network, without costing you anything significant.

So it is not really a great vector for commercialization. A much better approach to earning money would be to build a reliable and efficient network and charge for the access to this. Take your pick on whether this is last-mile hubs to the end-users, some sort of backbone, or both.

In the end, this is what it comes down to, efficient, secure and reliable data transport, from one point to another, with network-global end-to-end connectivity. Once you have that, everything else is just *protocol*, not servers.

**Comment by trosel** [comment]
> A much better approach to earning money would be to build a reliable and efficient network and charge for the access to this. 

Can you expand on what that might look like? My understanding is that there can't really be a "closed" network with Reticulum. There are "holes" in every Reticulum network because they can all communicate with each other. Right?

**Comment by markqvist** [comment]
That is right. Anyone could always add new segments to the network, and form new physical paths from one point to another, so nobody can have a monopoly on getting from one point to another.

One thing you can do is provide convenient and reliable access to a well-connected network over a geographical area with a model that your potential clients like, and makes it easier for them to use your service than setting up their own physical hardware on rooftops and hillsides.

For more concrete ideas or examples, I need to know what your actual goals are.

**Comment by trosel** [comment]
> One thing you can do is provide convenient and reliable access to a well-connected network over a geographical area with a model that your potential clients like, and makes it easier for them to use your service than setting up their own physical hardware on rooftops and hillsides

This could be run by a municipality too. Have the IT people maintain it with a few tax dollars :)

> For more concrete ideas or examples, I need to know what your actual goals are.

No goals as of now. This could probably be moved to a Discussion or something

**Comment by markqvist** [comment]
All good, thought you maybe had something specific in mind, and then it is hard to offer usable advice without knowing it. Yes, I'll move it to a discussion, probably better.

> This could be run by a municipality too.

Yes, Reticulum and LXMF is very suited to community-oriented management and maintenance.

Technological systems never exist in some sort of vacuum of platonic ideals, they always interface with peoples lives, and society at large, and as such plays a part in defining the boundaries of our possibilities for interaction and expression. In that context, I tried to put a lot of thought into which human consequences the design principles and technical decisions of these systems could have in the future.

I could go on about this topic for way too long, but in short, Reticulum and LXMF consciously tries to make a distributed and community-oriented management and maintenance model the most efficient one (both economically and socially). Such a model breeds distribution of power, and distribution of power breeds freedom for the individual. Freedom for the individual breeds new life and resources into the community. The aim is a positive feedback loop like that.
